torch/_prims/__init__.py

```diff
@@ -1802,18 +1802,33 @@ def collapse(a: Tensor, start: int, end: int) -> Tensor:
 
 # TODO: review stride logic
 def _cat_meta(tensors: Sequence[TensorLikeType], dim: int) -> TensorLikeType:
+    # Tensors must have the same number of dimensions
+    first_dims = tensors[0].ndim
```

Comment:
`cat` should explicitly inform the caller it doesn't accept an empty list of tensors:

```
RuntimeError: torch.cat(): expected a non-empty list of Tensors
```

Reply:
Aha, this is checked in the ref -- let's add a comment here, then